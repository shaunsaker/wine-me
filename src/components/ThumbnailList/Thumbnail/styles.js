import { StyleSheet } from 'react-native';

import styleConstants from '../../../styleConstants';

const IMAGE_SIZE = 120;

const styles = StyleSheet.create({
  image: {
    width: IMAGE_SIZE,
    height: IMAGE_SIZE,
    borderRadius: styleConstants.dimensions.borderRadius,
    overflow: 'hidden',
  },
});

export default styles;
