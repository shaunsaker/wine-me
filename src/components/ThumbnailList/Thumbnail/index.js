import React from 'react';
import PropTypes from 'prop-types';

import styles from './styles';

import RemoteImage from '../../RemoteImage';

const propTypes = {
  uri: PropTypes.string,
};

const defaultProps = {};

const Thumbnail = ({ uri }) => {
  return <RemoteImage source={{ uri }} style={styles.image} />;
};

Thumbnail.propTypes = propTypes;
Thumbnail.defaultProps = defaultProps;

export default Thumbnail;
