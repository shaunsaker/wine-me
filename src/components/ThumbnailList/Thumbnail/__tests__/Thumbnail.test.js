import React from 'react';
import renderer from 'react-test-renderer';

import Thumbnail from '..';

describe('Thumbnail', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<Thumbnail />);

      expect(component).toMatchSnapshot();
    });
  });
});
