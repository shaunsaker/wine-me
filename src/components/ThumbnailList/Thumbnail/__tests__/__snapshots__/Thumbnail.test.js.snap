// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail renders renders with minimum required props 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Array [
        Array [
          Object {
            "flex": 1,
            "height": 200,
          },
          Object {
            "borderRadius": 5,
            "height": 120,
            "overflow": "hidden",
            "width": 120,
          },
        ],
        Object {
          "overflow": "hidden",
        },
      ]
    }
  >
    <FastImageView
      borderRadius={5}
      onFastImageError={[Function]}
      onFastImageLoadEnd={[Function]}
      onFastImageLoadStart={[Function]}
      resizeMode="cover"
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
